src/legacy/core_plugins/kibana/public/management/sections/objects/components/object_view/form.tsx

```diff
@@ -0,0 +1,283 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Component } from 'react';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import {
+  forOwn,
+  indexBy,
+  cloneDeep,
+  isNumber,
+  isBoolean,
+  isPlainObject,
+  isString,
+  set,
+} from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  SimpleSavedObject,
+  SavedObjectsClientContract,
+} from '../../../../../../../../../core/public';
+import { castEsToKbnFieldTypeName } from '../../../../../../../../../plugins/data/public';
+import { SavedObjectLoader } from '../../../../../../../../../plugins/saved_objects/public';
+import { Field } from './field';
+import { ObjectField, FieldState, SubmittedFormData } from '../../types';
+
+interface FormProps {
+  object: SimpleSavedObject;
+  service: SavedObjectLoader;
+  savedObjectsClient: SavedObjectsClientContract;
+  editionEnabled: boolean;
+  onSave: (form: SubmittedFormData) => void;
+}
+
+interface FormState {
+  fields: ObjectField[];
+  fieldStates: Record<string, FieldState>;
+}
+
+export class Form extends Component<FormProps, FormState> {
+  constructor(props: FormProps) {
+    super(props);
+    this.state = {
+      fields: [],
+      fieldStates: {},
+    };
+  }
+
+  componentDidMount() {
+    const { object, service } = this.props;
+
+    const fields = Object.entries(object.attributes as Record<string, any>).reduce(
+      (objFields, [key, value]) => {
+        return [...objFields, ...recursiveCreateFields(key, value)];
+      },
+      [] as ObjectField[]
+    );
+    if ((service as any).Class) {
+      addFieldsFromClass((service as any).Class, fields);
+    }
+
+    this.setState({
+      fields,
+    });
+  }
+
+  render() {
+    const { editionEnabled, service } = this.props;
+    const { fields, fieldStates } = this.state;
+    const isValid = this.isFormValid();
+    return (
+      <EuiForm data-test-subj="savedObjectEditForm" role="form">
+        <div className="kuiVerticalRhythm">
+          {fields.map(field => (
+            <Field
+              key={`${field.type}-${field.name}`}
+              type={field.type}
+              name={field.name}
+              value={field.value}
+              state={fieldStates[field.name]}
+              disabled={!editionEnabled}
+              onChange={this.handleFieldChange}
+            />
+          ))}
+        </div>
+        <EuiFormRow fullWidth={true}>
```

Comment:
You don't need a `EuiFormRow` if you're not providing an input as a direct descendent or a `label`. It will just cause extraneous wrappers.